libcontainer/capabilities/capabilities.go

```diff
@@ -24,47 +24,69 @@ func init() {
 	}
 }
 
-// New creates a new Caps from the given Capabilities config.
+// New creates a new Caps from the given Capabilities config. Unknown Capabilities
+// or Capabilities that are unavailable in the current environment are ignored,
+// printing a warning instead.
 func New(capConfig *configs.Capabilities) (*Caps, error) {
 	var (
-		err  error
-		caps Caps
+		err      error
+		caps     Caps
+		ignored  []string
+		warnings = make([]string, 0)
 	)
 
-	if caps.bounding, err = capSlice(capConfig.Bounding); err != nil {
-		return nil, err
+	if caps.bounding, ignored = capSlice(capConfig.Bounding); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
 	}
-	if caps.effective, err = capSlice(capConfig.Effective); err != nil {
-		return nil, err
+	if caps.effective, ignored = capSlice(capConfig.Effective); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
 	}
-	if caps.inheritable, err = capSlice(capConfig.Inheritable); err != nil {
-		return nil, err
+	if caps.inheritable, ignored = capSlice(capConfig.Inheritable); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
 	}
-	if caps.permitted, err = capSlice(capConfig.Permitted); err != nil {
-		return nil, err
+	if caps.permitted, ignored = capSlice(capConfig.Permitted); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
 	}
-	if caps.ambient, err = capSlice(capConfig.Ambient); err != nil {
-		return nil, err
+	if caps.ambient, ignored = capSlice(capConfig.Ambient); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
 	}
 	if caps.pid, err = capability.NewPid2(0); err != nil {
 		return nil, err
 	}
 	if err = caps.pid.Load(); err != nil {
 		return nil, err
 	}
+	printWarnings(warnings)
 	return &caps, nil
 }
 
-func capSlice(caps []string) ([]capability.Cap, error) {
+func printWarnings(warnings []string) {
+	if len(warnings) == 0 {
+		return
+	}
+	var unknownCaps = map[string]struct{}{}
+	for _, c := range warnings {
+		if _, ok := unknownCaps[c]; !ok {
+			logrus.WithField("capability", c).Warn("ignoring unknown or unavailable capability")
```

Comment:
Only consideration I had here was if we wanted to pass a logger to `New()`, instead of always using the default logrus logger (in case this is used elsewhere as library)

Reply:
Also not sure if it must be a WARN level, or if we think INFO (possibly with "warning" in the log message) to reduce noise

Reply:
I'd rather have a single message listing all the capabilities, maybe something like `Warning: unknown capabilities [CAP_ONE CAP_TWO ...] are ignored`.